src/lib/base.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import print_function
+
+import json
+import os
+import re
+import sys
+
+from jinja2 import StrictUndefined, Template, TemplateSyntaxError, UndefinedError
+from slackclient import SlackClient
+
+
+def template_str(text, variables):
+    ''' Return a templated text '''
+    # Merge ENV with variables passed
+    os_env = {"ENV": os.environ}
+    variables.update(os_env)
+
+    try:
+        return Template(text, undefined=StrictUndefined).render(variables)
+    except TemplateSyntaxError as syntax_error:
+        fail_unless(False, "Template syntax error. Template string: '{}'\n.{}".format(text, syntax_error))
+    except UndefinedError as undefined_error:
+        fail_unless(False, "Undefined variable. Template string: '{}'\n.{}".format(text, undefined_error))
+    except TypeError as type_error:
+        fail_unless(False, "Type error. Template string: '{}'\n.{}".format(text, type_error))
+
+def template_with_regex(text, regex):
+    extra_env = {}
+    if regex and regex.groupdict():
+        extra_env = {"regex": regex.groupdict()}
+    elif regex and regex.groups():
+        extra_env = {"regex": regex.groups()}
+    return template_str(text, extra_env)
+
+def write_to_file(content, output_file):
+    ''' write content to output_file'''
+    output_file_fd = open(output_file, 'w')
+    print(content, file=output_file_fd)
+
+
+def fail_unless(condition, msg):
+    """If condition is not True print msg and exit with status code 1"""
+    if not condition:
+        print("{}".format(msg), file=sys.stderr)
+        exit(1)
+
+
+class Base(object):
+    """Slack Concourse resource implementation"""
+
+    def __init__(self, **kwargs):
+        self.slack_client = SlackClient(kwargs.get("slack_token"))
```

Comment:
instead of initialising the slack client directly here, it would be better to inject the dependency. Doing so, would make it easier to test it